### PR TITLE
CAA fixes

### PR DESCRIPTION
**Pull request by mwiencek**
Addresses several CAA issues reported by Ruben Opdebeeck (@ROpdebee) via email. All credit goes to him for researching these and proposing solutions.

I've tested the changes locally with the help of some additions to contrib/ssssss.psgi.

### CODE REVIEW

**Comment by yvanzo** [comment]
@mwiencek: Should tests be updated?

**Comment by mwiencek** [comment]
@yvanzo Yes, I've pushed some fixes, let's see if they pass now.

**Review by reosarevok — APPROVED** [review]
LGTM but didn't test. First two fixes are trivial in any case. Third is a lot more complex and I'm not fully aware of how every step of the way works for CAA uploads, but seems sane and if
@ROpdebee has checked it too I'm ok with it :)

**Comment by ROpdebee** [comment]
I haven't tested these changes locally either, and I don't have a lot of experience with MBS, but this looks good to me.

File: contrib/ssssss.psgi [hunk 1]
```diff
@@ -80,17 +93,53 @@ sub handle_options
...
+        $response->content_type('text/xml');
+        $response->body(q{<?xml version='1.0' encoding='UTF-8'?>
+            <Error><Code>BucketAlreadyExists</Code></Error>});
+        return $response;
+    } else {
+        create_bucket($request->path_info);
+    }
 
     return $request->new_response(204);
 }
```
Comment (yvanzo) [inline contrib/ssssss.psgi:145]: Can we log these errors using Sentry? Or do we have access to them elsewhere already?

**Review by yvanzo — APPROVED** [review]
Reviewed closely but could not test; Hopefully there are mechanized tests.

Thanks again to @ROpdebee for deeply investigating this issue!

File: contrib/ssssss.psgi (hunk 1 above)
Comment (mwiencek) [reply contrib/ssssss.psgi:145]: We'd need to add some methods to MusicBrainz::Errors to allow logging arbitrary messages to Sentry. I don't have the means to test that right now, but I could look into it after the schema change.

(But we wouldn't want to log all `BucketAlreadyExists` errors; only cases where the ownership is wrong. And that should rarely ever happen, hopefully, other than the one case we know about.)

**Comment by ROpdebee** [comment]
MBS-11624

**Pull request merged**